docs/permissions/plugin-authors/02-adding-a-basic-permission-check.md

```diff
@@ -134,30 +141,47 @@ export async function createRouter(
   // ...
 ```
 
-Pass the `permissions` object to the plugin in `packages/backend/src/plugins/todolist.ts`:
+Pass the `auth` and `permissions` objects to the plugin in `plugins/todo-list-backend/src/plugin.ts`:
 
-```ts title="packages/backend/src/plugins/todolist.ts"
-import { DefaultIdentityClient } from '@backstage/plugin-auth-node';
-import { createRouter } from '@internal/plugin-todo-list-backend';
-import { Router } from 'express';
-import { PluginEnvironment } from '../types';
+```ts title="plugins/todo-list-backend/src/plugin.ts"
+import { coreServices, createBackendPlugin } from '@backstage/backend-plugin-api';
+import { createRouter } from './service/router';
 
-export default async function createPlugin({
-  logger,
-  discovery,
-  /* highlight-add-next-line */
-  permissions,
-}: PluginEnvironment): Promise<Router> {
-  return await createRouter({
-    logger,
-    identity: DefaultIdentityClient.create({
-      discovery,
-      issuer: await discovery.getExternalBaseUrl('auth'),
-    }),
-    /* highlight-add-next-line */
-    permissions,
-  });
-}
+export const exampleTodoListPlugin = createBackendPlugin({
+  pluginId: 'todolist',
+  register(env) {
+    env.registerInit({
+      deps: {
+        logger: coreServices.logger,
+        /* highlight-add-start */
+        auth: coreServices.auth,
+        permissions: coreServices.permissions,
+        /* highlight-add-end */
+        httpAuth: coreServices.httpAuth,
+        httpRouter: coreServices.httpRouter,
+      },
+      /* highlight-remove-next-line */
+      async init({ logger, httpAuth, httpRouter }) {
+      /* highlight-add-next-line */
+      async init({ logger, auth, httpAuth, httpRouter, permissions }) {
+        httpRouter.use(
+          await createRouter({
+            logger,
+            /* highlight-add-start */
+            auth,
+            permissions,
+            /* highlight-add-end */
+            httpAuth,
+          }),
+        );
+        httpRouter.addAuthPolicy({
```

Comment:
is this part of the rest of the tutorial or can it be skipped to get less unrelated code?

Reply:
This code is based on the previous setup step, which uses [this](https://github.com/backstage/backstage/blob/8f85ab3b04e3311171ca281d722c5177a5bba576/plugins/example-todo-list-backend/src/plugin.ts#L44) example plugin code as the base for the tutorial. I just used that as the base for the snippets here too.